jetty-proxy/src/test/java/org/eclipse/jetty/proxy/ProxyServletTest.java

```diff
@@ -549,6 +612,79 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
             Matchers.equalTo("localhost:" + serverConnector.getLocalPort()));
     }
 
+    @ParameterizedTest
+    @MethodSource("subclassesWithProtocols")
+    public void testInheritance(ProxyServlet derivedProxyServlet, String protocol) throws Exception
```

Comment:
Please no.
It's impossible to read this test and understand what's doing without jumping in 3 other different places in the code.
Don't use mock anywhere, not needed and too brittle.
Don't use `@ParameterizedTest`, not needed here.

Reply:
Alt-Space in Intellij and I can see the implementation of all 3 of those other methods without leaving the test. Having code in various places isn't a problem IHMO, especially when that code does only 1 thing. I want tests that are easy to read and understand the intent. Cruft obscures that and makes the test harder to understand. This is your guys party, and I will change this if you insist. 

If this test isn't parameterized, then compatibility with all existing subclasses isn't tested. In which case, the test can be deleted.

Reply:
Oh the irony of somebody using some intellij key sequence in discussion with @sbordet, who is normally the one to say that ctrl-rightalt-elbow-headbutt will solve your problems!  @travisspencer you just made my day :)

But seriously,   I too am not a fan of mock in test harness and in this case I find the method names confusing as it looks like they will return some third party mock library generated test classes.  But they don't, they just return simple instantiations of the real classes.  So in this case I too think it would be simpler to inline them and avoid the confusion of a "mock" name in a method.

Reply:
I shift the code around as @sbordet suggested. My consolation is that I made your day, @gregw .